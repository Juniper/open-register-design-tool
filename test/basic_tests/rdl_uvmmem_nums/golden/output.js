/*   Ordt 190725.01 autogenerated file 
 *   Input: ./rdl_uvmmem_nums/test.rdl
 *   Parms: ./rdl_uvmmem_nums/test.parms
 *   Date: Sun Nov 17 17:01:11 EST 2019
 */

register_set top "Registers for top" {

  address = 0x0;
  register_width = 32;
  register sr1 "sr1 register" {
    address = 0x0;
    integer fld1[32] "fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register wr1 "wr1 register" {
    address = 0x100;
    register_width = 128;
    integer fld8[16] "fld8 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld7[16] "fld7 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld6[16] "fld6 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld5[16] "fld5 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld4[16] "fld4 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld3[16] "fld3 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld2[16] "fld2 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld1[16] "fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register sr_repeat1 "sr_repeat1 register" {
    address = 0x200;
    integer fld1[32] "fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register sr_repeat2 "sr_repeat2 register" {
    address = 0x204;
    integer fld1[32] "fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register sr_repeat3 "sr_repeat3 register" {
    address = 0x208;
    integer fld1[32] "fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register srmem1 "srmem1 register" {
    address = 0x2000;
    repeat = 100;
    integer fld1[32] "fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register wrmem1 "wrmem1 register" {
    address = 0x20000;
    register_width = 128;
    repeat = 300;
    integer fld8[16] "fld8 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld7[16] "fld7 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld6[16] "fld6 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld5[16] "fld5 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld4[16] "fld4 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld3[16] "fld3 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld2[16] "fld2 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer fld1[16] "fld1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register_set_size = 0x212c0;
};

