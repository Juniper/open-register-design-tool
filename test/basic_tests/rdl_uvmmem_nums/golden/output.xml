<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 190725.01 autogenerated file 
       Input: ./rdl_uvmmem_nums/test.rdl
       Parms: ./rdl_uvmmem_nums/test.parms
       Date: Sun Nov 17 17:08:31 EST 2019
 -->

<map version="190725.01">
  <id>top</id>
  <baseaddr>0</baseaddr>
  <shorttext>top registers</shorttext>
  <reg>
    <id>sr1</id>
    <shorttext>sr1 register</shorttext>
    <baseaddr>0x0</baseaddr>
    <width>32</width>
    <field>
      <id>fld1</id>
      <shorttext>fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>32</width>
    </field>
    <access>RW</access>
  </reg>
  <reg>
    <id>wr1</id>
    <shorttext>wr1 register</shorttext>
    <baseaddr>0x100</baseaddr>
    <width>128</width>
    <field>
      <id>fld1</id>
      <shorttext>fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld2</id>
      <shorttext>fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>16</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld3</id>
      <shorttext>fld3 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>32</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld4</id>
      <shorttext>fld4 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>48</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld5</id>
      <shorttext>fld5 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>64</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld6</id>
      <shorttext>fld6 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>80</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld7</id>
      <shorttext>fld7 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>96</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld8</id>
      <shorttext>fld8 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>112</lowidx>
      <width>16</width>
    </field>
    <access>RW</access>
  </reg>
  <reg>
    <id>sr_repeat1</id>
    <shorttext>sr_repeat1 register</shorttext>
    <baseaddr>0x200</baseaddr>
    <width>32</width>
    <field>
      <id>fld1</id>
      <shorttext>fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>32</width>
    </field>
    <access>RW</access>
  </reg>
  <reg>
    <id>sr_repeat2</id>
    <shorttext>sr_repeat2 register</shorttext>
    <baseaddr>0x204</baseaddr>
    <width>32</width>
    <field>
      <id>fld1</id>
      <shorttext>fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>32</width>
    </field>
    <access>RW</access>
  </reg>
  <reg>
    <id>sr_repeat3</id>
    <shorttext>sr_repeat3 register</shorttext>
    <baseaddr>0x208</baseaddr>
    <width>32</width>
    <field>
      <id>fld1</id>
      <shorttext>fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>32</width>
    </field>
    <access>RW</access>
  </reg>
  <reg>
    <id>srmem1</id>
    <shorttext>srmem1 register</shorttext>
    <baseaddr>0x2000</baseaddr>
    <width>32</width>
    <reps>100</reps>
    <stride>0x4</stride>
    <field>
      <id>fld1</id>
      <shorttext>fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>32</width>
    </field>
    <access>RW</access>
  </reg>
  <reg>
    <id>wrmem1</id>
    <shorttext>wrmem1 register</shorttext>
    <baseaddr>0x20000</baseaddr>
    <width>128</width>
    <reps>300</reps>
    <stride>0x10</stride>
    <field>
      <id>fld1</id>
      <shorttext>fld1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld2</id>
      <shorttext>fld2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>16</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld3</id>
      <shorttext>fld3 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>32</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld4</id>
      <shorttext>fld4 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>48</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld5</id>
      <shorttext>fld5 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>64</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld6</id>
      <shorttext>fld6 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>80</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld7</id>
      <shorttext>fld7 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>96</lowidx>
      <width>16</width>
    </field>
    <field>
      <id>fld8</id>
      <shorttext>fld8 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>112</lowidx>
      <width>16</width>
    </field>
    <access>RW</access>
  </reg>
</map>
